plugins/woocommerce/src/Internal/Orders/CouponsController.php

```diff
@@ -79,12 +79,16 @@ public function add_coupon_discount( array $post_variables ): object {
 		$order->calculate_taxes( $calculate_tax_args );
 		$order->calculate_totals( false );
 
-		$result = $order->apply_coupon( wc_format_coupon_code( wp_unslash( $coupon ) ) ); // phpcs:ignore WordPress.Security.NonceVerification.Missing, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		$code   = wc_format_coupon_code( wp_unslash( $coupon ) ); // phpcs:ignore WordPress.Security.NonceVerification.Missing, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		$result = $order->apply_coupon( $code );
 
 		if ( is_wp_error( $result ) ) {
 			throw new Exception( html_entity_decode( wp_strip_all_tags( $result->get_error_message() ) ) );
 		}
 
+		// translators: %s coupon code.
+		$order->add_order_note( sprintf( __( 'Coupon applied: "%s".', 'woocommerce' ), $code ), 0, true );
```

Comment:
Similar to an earlier note:

```suggestion
		$order->add_order_note( esc_html( sprintf( __( 'Coupon applied: "%s".', 'woocommerce' ), $code ), 0, true ) );
```

(Guards against HTML in translations, and potentially against HTML in `$code`—even though we don't expect that in most cases.)